activesupport/lib/active_support/railtie.rb

```diff
@@ -7,6 +7,17 @@ class Railtie < Rails::Railtie # :nodoc:
 
     config.eager_load_namespaces << ActiveSupport
 
+    # Use_authenticated_message_encryption = true, turns on 'aes-256-gcm' as
+    # the default cipher. If use_authenticated_message_encryption = false, then
+    # the default_cipher used is 'aes-256-cbc'.
+    config.active_support.use_authenticated_message_encryption = false
+    
+    if config.active_support.use_authenticated_message_encryption
+      ActiveSupport::MessageEncryptor.use_authenticated_message_encryption = true
+    else
+      ActiveSupport::MessageEncryptor.use_authenticated_message_encryption = false
+    end
```

Comment:
This all needs to be wrapped in an `initializer` (with a descriptive name) so it's run when the app boots and not when this file is parsed.

This Railtie subclass is marked as `# :nodoc` meaning it won't show up in documentation, so just skip the comments here.

I'd rewrite to this within the `initializer`:

```ruby
if config.active_support.respond_to?(:use_authenticated_message_encryption)
  ActiveSupport::MessageEncryptor.use_authenticated_message_encryption =
    config.active_support.use_authenticated_message_encryption
end
```

We have to use `respond_to?` because `config.active_support`, an instance of `ActiveSupport::OrderedOptions`, raises if the key is unassigned.